numpy/core/_exceptions.py

```diff
@@ -117,6 +117,16 @@ def __str__(self):
 # Exception used in shares_memory()
 @set_module('numpy')
 class TooHardError(RuntimeError):
+    """max_work was exceeded.
+
+    This is raised whenever the maximum number of candidate solutions 
+    to consider specified by the ``max_work`` parameter exceeds.
+    Since checking whether two arrays share memory is NP-complete 
+    you have may considered setting up max_work to a finite number 
```

Comment:
This phrasing is a bit confusing to me - do you mean to suggest that people should consider setting up max_work to a finite number, or are you saying they did this and that's why the error occurred? If it's the first option, I would write
```suggestion
    you may consider setting up max_work to a finite number 
```

Reply:
Hello, @melissawm. I am trying to say that the error occurred because a finite number was given to `max_work` attribute. What do you think about changing the sentence to `Setting up max_work to a finite number may have caused the operation to fail` ? Does it make sense?
Additionally, would you like me to open a new PR according to Charles's recommendations?

Reply:
@angellicacardozo @melissawm Consider the following wording:
```suggestion
Assigning a finite number to max_work may have caused the operation 
to fail.
```